sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -100,10 +119,12 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
         HttpHeader operationLocationHeader = response.getHeaders().get(operationLocationHeaderName);
         HttpHeader locationHeader = response.getHeaders().get(PollingConstants.LOCATION);
         if (operationLocationHeader != null) {
-            pollingContext.setData(operationLocationHeaderName, operationLocationHeader.getValue());
+            pollingContext.setData(operationLocationHeaderName,
+                getAbsolutePath(operationLocationHeader.getValue(), endpoint, LOGGER));
         }
         if (locationHeader != null) {
-            pollingContext.setData(PollingConstants.LOCATION, locationHeader.getValue());
+            pollingContext.setData(PollingConstants.LOCATION,
+                getAbsolutePath(locationHeader.getValue(), endpoint, LOGGER));
         }
         pollingContext.setData(PollingConstants.HTTP_METHOD, response.getRequest().getHttpMethod().name());
         pollingContext.setData(PollingConstants.REQUEST_URL, response.getRequest().getUrl().toString());
```

Comment:
Do we need to override getResult() to make get final result url absolute path as well?

Reply:
`response.getRequest().getUrl()` is from a working request and must be absolute.

Reply:
updated